test/UID2.Client.Test/SharingClientTests.cs

```diff
@@ -0,0 +1,420 @@
+
+            DecryptAndAssertSuccess(advertisingToken, tokenVersion);
+        }
+
+        [Theory]
+        [InlineData(IdentityScope.UID2, TokenVersion.V3)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V3)]
+        [InlineData(IdentityScope.UID2, TokenVersion.V4)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V4)]
+        private void PhoneTest(IdentityScope identityScope, TokenVersion tokenVersion)
+        {
+            var refreshResult = _client.RefreshJson(KeySharingResponse(new[] { MASTER_KEY, SITE_KEY }, identityScope));
+            Assert.True(refreshResult.Success);
+
+            const string rawUidPhone = "BEOGxroPLdcY7LrSiwjY52+X05V0ryELpJmoWAyXiwbZ";
+            var advertisingToken = AdvertisingTokenBuilder.Builder().WithVersion(tokenVersion).WithScope(identityScope).WithRawUid(rawUidPhone).Build();
+
+            var res = _client.DecryptTokenIntoRawUid(advertisingToken);
+            Assert.True(res.Success);
+            Assert.Equal(rawUidPhone, res.Uid);
+            Assert.Equal((int)tokenVersion + 2, res.AdvertisingTokenVersion);
+            Assert.Equal(IdentityType.Phone, res.IdentityType);
+        }
+
+        [Theory]
+        [InlineData(TokenVersion.V2)]
+        [InlineData(TokenVersion.V3)]
+        [InlineData(TokenVersion.V4)]
+        private void LegacyResponseFromOldOperator(TokenVersion tokenVersion)
+        {
+            var refreshResult = _client.RefreshJson(TestData.KeySharingResponse(new[] { MASTER_KEY, SITE_KEY }));
+            Assert.True(refreshResult.Success);
+
+            var advertisingToken = AdvertisingTokenBuilder.Builder().WithVersion(tokenVersion).Build();
+
+            DecryptAndAssertSuccess(advertisingToken, tokenVersion);
+
+        }
+
+        [Theory]
+        [InlineData(IdentityScope.UID2, TokenVersion.V2)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V2)]
+        [InlineData(IdentityScope.UID2, TokenVersion.V3)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V3)]
+        [InlineData(IdentityScope.UID2, TokenVersion.V4)]
+        [InlineData(IdentityScope.EUID, TokenVersion.V4)]
+        private void TokenGeneratedInTheFutureLegacyClient(IdentityScope identityScope, TokenVersion tokenVersion)
+        {   //similar to BidstreamClientTests.TokenGeneratedInTheFutureLegacyClient, but uses KeySharingResponse and Decrypt without domain parameter
+            UID2Client client = new("endpoint", "authkey", CLIENT_SECRET, identityScope);
+
+            var refreshResult = client.RefreshJson(KeySharingResponse(new[] { MASTER_KEY, SITE_KEY }, identityScope));
+            Assert.True(refreshResult.Success);
+
+            var advertisingToken = AdvertisingTokenBuilder.Builder().WithVersion(tokenVersion).WithScope(identityScope).WithGenerated(DateTime.UtcNow.AddMinutes(31)).Build();
```

Comment:
AddDays(30).AddMinutes(1)?

Reply:
changed to AddDays(99) since any future date is supposed to work for legacy client (for now)